stl/CMakeLists.txt

```diff
@@ -458,11 +458,11 @@ function(add_stl_dlls D_SUFFIX THIS_CONFIG_DEFINITIONS THIS_CONFIG_COMPILE_OPTIO
     target_compile_options(msvcp${D_SUFFIX}_eha_objects PRIVATE "${THIS_CONFIG_COMPILE_OPTIONS};${GL_FLAG};/EHa")
 
     add_library(msvcp${D_SUFFIX} SHARED)
-    target_link_libraries(msvcp${D_SUFFIX} PRIVATE msvcp${D_SUFFIX}_eha_objects msvcp${D_SUFFIX}_objects msvcp${D_SUFFIX}_init_objects "${TOOLSET_LIB}/vcruntime${D_SUFFIX}.lib" "${TOOLSET_LIB}/msvcrt${D_SUFFIX}.lib" "ucrt${D_SUFFIX}.lib")
+    target_link_libraries(msvcp${D_SUFFIX} PRIVATE msvcp${D_SUFFIX}_eha_objects msvcp${D_SUFFIX}_objects msvcp${D_SUFFIX}_init_objects "${TOOLSET_LIB}/vcruntime${D_SUFFIX}.lib" "${TOOLSET_LIB}/msvcrt${D_SUFFIX}.lib" "ucrt${D_SUFFIX}.lib" "ole32.lib" "delayimp.lib")
     set_target_properties(msvcp${D_SUFFIX} PROPERTIES ARCHIVE_OUTPUT_NAME "msvcp140_base${D_SUFFIX}${VCLIBS_SUFFIX}")
     set_target_properties(msvcp${D_SUFFIX} PROPERTIES ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
     set_target_properties(msvcp${D_SUFFIX} PROPERTIES OUTPUT_NAME "msvcp140${D_SUFFIX}${VCLIBS_SUFFIX}")
-    target_link_options(msvcp${D_SUFFIX} PRIVATE "${THIS_CONFIG_LINK_OPTIONS}")
+    target_link_options(msvcp${D_SUFFIX} PRIVATE "${THIS_CONFIG_LINK_OPTIONS}" "-delayload:ole32.dll")

```

Comment:
we don't use delay loading in the STL, as it's so, so, so easy to accidentally create compatibility bugs. Instead, we have code in a satellite dll that has a global (to that dll) table of function pointers, and functions to load said table from whatever dll's the functions come from (using load library/getmodulehandle/etc) and return error information if that was unsuccessful.

You can find examples of this pattern in stl/src/atomic_wait.cpp (see _Wait_functions_table) and stl/src/tzdb.cpp (see _Icu_functions_table).

It must be a new sat dll unless there's a sat dll which we have added but has not been released yet, this is because even additions to existing dlls break ABI for us, since plugins using the STL need to be able to be loaded into a process that has already loaded an earlier version.

Reply:
That all sounds like delay loading by another name.  But happy to use another technique.

Reply:
Note that although it's possible to add a satellite DLL, that's an extremely burdensome process and I don't think it should be necessary here (it's necessary only when we need shared global state).